### PR TITLE
Add never delete user flag

### PR DESCRIPTION
**Pull request by Jim3Things**

### CODE REVIEW

File: internal/services/frontend/DBUsers.go [hunk 1]
```diff
@@ -113,13 +119,16 @@ func (m *DBUsers) Update(u *pb.User, match int64) (int64, error) {
...
+    // Update the entry, retaining the fields from the old version that are
+    // immutable
     entry := &pb.UserInternal{
         User:                 &pb.User{
             Name:                 u.Name,
             PasswordHash:         u.PasswordHash,
             UserId:               u.UserId,
             Enabled:              u.Enabled,
             AccountManager:       u.AccountManager,
+            NeverDelete:          old.User.NeverDelete,
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/DBUsers.go:131]: I'm afraid it's not clear to my why the UserId and Name are not also considered immutable?

File: internal/services/frontend/users_test.go [hunk 2]
```diff
@@ -476,13 +482,27 @@ func TestUsersList(t *testing.T) {
...
+	match := knownNames
+	match[baseURI + admin] = baseURI + admin
+
+	// .. this involves converting the set of keys to an array for matching
+	keys := make([]string, 0, len(match))
+	for k := range match {
+		keys = append(keys, k)
+	}
+
+	assert.ElementsMatchf(t, keys, names[1:], "elements did not match\nReturned Value: %s\nMatch Values: %v", s, keys)
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/users_test.go:505]: Cute.

File: internal/services/frontend/users_test.go [hunk 3]
```diff
@@ -499,13 +519,9 @@ func TestUsersListNoPriv(t *testing.T) {
...
 
 	response = doHTTP(request, response.Cookies())
-	body, err := getBody(response)
+	_, err := getBody(response)
 
 	assert.Nilf(t, err, "Failed to read body returned from call to handler for route %v: %v", userURI, err)
-
-	t.Logf("[%s]: SC=%v, Content-Type='%v'\n", userURI, response.StatusCode, response.Header.Get("Content-Type"))
-	t.Log(string(body))
-
```
Comment (MikeCJohnsonUk) [inline internal/services/frontend/users_test.go:508]: I take it this was essentially debug code?

File: internal/services/frontend/DBUsers.go (hunk 1 above)
Comment (Jim3Things) [reply internal/services/frontend/DBUsers.go:131]: Good catch.  Missed that the first time around.  Fixed in next push.

File: internal/services/frontend/users_test.go (hunk 2 above)
Comment (Jim3Things) [reply internal/services/frontend/users_test.go:505]: Yeah, I missed a lot of the weirder assert options at the start...

File: internal/services/frontend/users_test.go (hunk 3 above)
Comment (Jim3Things) [reply internal/services/frontend/users_test.go:508]: Yes.

**Pull request merged**